### PR TITLE
fix(clerk-js): Navigate to `/choose` when signing out during multi session

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

https://github.com/user-attachments/assets/66e41cbd-2059-4808-9ae8-5dc3ccc58796

Fixes SDKI-671

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/components/UserButton/useMultisessionActions.tsx [hunk 1]
```diff
@@ -27,8 +27,11 @@ export const useMultisessionActions = (opts: UseMultisessionActionsParams) => {
     if (otherSessions.length === 0) {
       return signOut(opts.navigateAfterSignOut);
     }
-    return signOut(opts.navigateAfterMultiSessionSingleSignOut, { sessionId: session.id }).finally(() =>
-      card.setIdle(),
+    return signOut(opts.navigateAfterMultiSessionSingleSignOut, { sessionId: session.id }).finally(
+      () =>
+        void navigate(`${opts.signInUrl}/choose`).finally(() => {
+          card.setIdle();
+        }),
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/components/UserButton/useMultisessionActions.tsx:34]: @alexcarpenter Wouldn't this redirect twice ? Once when `navigateAfterMultiSessionSingleSignOut` is called and then to `/choose` ?

I think this logic should go inside `navigateAfterMultiSessionSingleSignOut`

**Review by panteliselef — CHANGES_REQUESTED** [review]
I'm just being cautious here.

File: packages/clerk-js/src/ui/components/UserButton/useMultisessionActions.tsx (hunk 1 above)
Comment (alexcarpenter) [reply packages/clerk-js/src/ui/components/UserButton/useMultisessionActions.tsx]: @panteliselef did a bit of pairing with @dstaley which can be seen here https://github.com/clerk/javascript/pull/4203/commits/0026ca54189d15f0c4fbab913a90f55cb342796a

Let me know what you think. Definitely being cautious welcomed 😄 added a few tests to validate the updates.

File: packages/clerk-js/src/core/clerk.ts [hunk 2]
```diff
@@ -969,7 +969,7 @@ export class Clerk implements ClerkInterface {
 
   public buildAfterMultiSessionSingleSignOutUrl(): string {
     if (!this.#options.afterMultiSessionSingleSignOutUrl) {
-      return this.buildAfterSignOutUrl();
+      return this.environment ? `${this.environment.displayConfig.signInUrl}/choose` : this.buildAfterSignOutUrl();
```
Comment (panteliselef) [inline packages/clerk-js/src/core/clerk.ts:972]: ❓ if the signInUrl is defined shouldn't that be used instead of the values retrieved the environment ?

**Review by panteliselef — APPROVED** [review]

**Pull request merged**